### PR TITLE
Use 'tree' style stat names, and riak_core_stat_q

### PR DESCRIPTION
**Pull request by russelldb**
Use riak core stat query for riak_kv_stat get_stats
Add new read repair dimensioned stat aggregations to stat blob

Depends on riak_core pr - https://github.com/basho/riak_core/pull/252

### CODE REVIEW

File: src/riak_kv_stat_bc.erl [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+                        orddict:store(Key, [{count, C+C0}, {one, O+O0}], Acc)
+                end,
+                orddict:new(),
+                Stats).
+
+key_from_fields(Name, Fields) ->
+    Key = [element(N, Name) || N <- Fields],
+    join(Key).
+
+recursive_join(BaseName, Aggregates) ->
```
Comment (beerriot) [inline src/riak_kv_stat_bc.erl]: The name `recursive_join` was a little vague for me. Maybe a comment explaining how it's flattening the nested stat tree, creating a flat list of stats with names prepended by `BaseName_`?

File: src/riak_kv_stat_bc.erl [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+    L = orddict:fold(fun(K, V, Acc) when not is_list(V) ->
+                             [{join([BaseName, K]), V}|Acc];
+                        (K, V, Acc)  ->
+                             [recursive_join(join([BaseName, K]), V)|Acc]
+                     end,
+                     [],
+                     Aggregates),
+    lists:flatten(L).
```
Comment (beerriot) [inline src/riak_kv_stat_bc.erl]: It seems like it shouldn't be necessary to call `lists:flatten/1` on every recursion of this function … it should only be needed once at the top level (in `aggregate/3`?).

File: src/riak_kv_stat.erl [hunk 3]
```diff
@@ -443,71 +263,16 @@ register_stat(Name, histogram) ->
     {SampleType, SampleArgs} = get_sample_type(Name),
     folsom_metrics:new_histogram(Name, SampleType, SampleArgs).
 
+%% @doc the histogram sample type may be set in app.config
+%% use key `stat_sample_type` in the `riak_kv` section. Or the
```
Comment (beerriot) [inline src/riak_kv_stat.erl]: These are markdown-style literal escapes, not edoc-style.

**Comment by russelldb** [comment]
Pushed a commit that addresses review comments.

**Comment by beerriot** [comment]
Looks great, +1!

**Pull request merged**